external/fv3fit/tests/data/test_netcdf.py

```diff
@@ -0,0 +1,34 @@
+from pathlib import Path
+
+import fv3fit.data
+import numpy as np
+import vcm
+
+
+def test_NetcdfDirLoader(tmp_path: Path):
+    path = tmp_path / "a.nc"
+    cache = tmp_path / ".cache"
+    ds = vcm.cdl_to_dataset(
+        """
+    netcdf A {
+        dimensions:
+            sample = 4;
+            z = 1;
+        variables:
+            float a(sample, z);
+        data:
+            a = 0, 1, 2, 3;
+    }
+    """
+    )
+
+    ds.to_netcdf(path.as_posix())
+    loader = fv3fit.data.NetcdfDirLoader(tmp_path.as_posix())
+    tfds = loader.open_tfdataset(cache.as_posix(), ["a"])
+    for data in tfds.as_numpy_iterator():
+        a = data["a"]
+    np.testing.assert_almost_equal(ds["a"], a)
```

Comment:
optional nit: you should be able to assert it's identically equal in this case. Would cover any issues like accidentally casting between float32 and float64, but maybe you don't want to cover that.